### ISSUE TITLE
Upcoming release.

### ISSUE THREAD

**Issue by Trancever** [callstack/react-native-fbads#141]
@erandagan @Suraj-Tiwari 
Hi guys, I would like to discuss what should be done before we release new version of fbads. Is there anything really important that we should address before release?

**Comment by Suraj-Tiwari** [comment]
Yeah. fully test it and review it, i slightly checked it but due to busy
schedules i wasn't able to review it completely

**Comment by Suraj-Tiwari** [comment]
This library isn't like we can go with unstable version. We should make it
stable enough so that won't cause any problems with any project. Random
crashes can affect app reputation.

**Comment by erandagan** [comment]
I referred to API stability, not to code stability. The API may break in future versions, it does not mean that the code itself will crash your app.

**Comment by Suraj-Tiwari** [comment]
I was referring to code stability. :P

**Comment by erandagan** [comment]
Cool. Anyway, I think that the following is required:
- A list of breaking changes. This can be copied from my own pull requests which introduced them.
- A note stating this library is undergoing some changes at the moment, which may lead to breaking some APIs

Also, we can consider making this an RC release so we can continue introducing breaking changes without breaking any dependencies for the users, until we have something stable ready and we can release it as a major breaking change.

If we'd go in the RC path i'd also add a note about it in the README stating that any testing of RC releases is very welcome and helpful.

**Comment by Suraj-Tiwari** [comment]
this will be good. releasing it as beta

**Comment by Suraj-Tiwari** [comment]
@erandagan for npm release we can just publish distribution for better performance rather than source code. How is it?

**Comment by erandagan** [comment]
We can go ahead and release a beta / RC now.

**Comment by Suraj-Tiwari** [comment]
Yeah everything is ready to released.

**Comment by Jobeso** [comment]
Hey guys, you are developing an awesome library here. I am using an older version of it in production. When trying out the newest beta I encountered a few things. Where should feedback to the newest beta be posted and are there any docs about the changes yet?

**Comment by erandagan** [comment]
Thanks. To use the latest version checkout the master branch (the beta branch is actually outdated) which also has up to date documentation.

If you have feedback or you encounter any bugs, just submit an issue in this repo.

**Comment by Suraj-Tiwari** [comment]
@Trancever @erandagan all pr are meged i've tested (without last pr).
Proceed to release `rc v6`?

**Comment by Trancever** [comment]
@Suraj-Tiwari @erandagan If we are happy with current status of the lib then yes, but if we want to introduce more breaking changes in near future i would rather wait with the release.

**Comment by Suraj-Tiwari** [comment]
I'm busy for this week only after that I'll have time.
If there are any plans for implementing Tests and some pre-built
implementation/view to make process fast and less complex. That will be
super efficient i guess.
What you think?

**Comment by erandagan** [comment]
@Trancever I think it's a good idea to release an RC so people can give this a shot and post their feedback, it will allow us to iterate quickly towards a stable release with a stable API.

I don't expect breaking changes for now, except maybe in the TriggerableView area?

**Comment by Trancever** [comment]
@erandagan @Suraj-Tiwari Version 6.0.0-RC landed on npm :)

**Comment by Suraj-Tiwari** [comment]
Did you merged my pr? or it's still pending?

**Comment by Suraj-Tiwari** [comment]
Anyway that's great I'll check out this one.

**Comment by erandagan** [comment]
@Trancever 
How about we submit a 6.0.0-RC2 release with the fixed dependencies and transpiled TS files?

**Comment by Suraj-Tiwari** [comment]
This one is fully tested or unlike previous build?

**Comment by Suraj-Tiwari** [comment]
@erandagan @Trancever 
Recently I've seen Some issue related to Rc-release With some common issue, I think we should have a checklist before releasing.

**Comment by erandagan** [comment]
Not really a checklist, but tighter CI integration. I've opened an issue for that yesterday.

**Comment by Suraj-Tiwari** [comment]
Anyone working on it?

**Comment by erandagan** [comment]
I don't think so

**Comment by Suraj-Tiwari** [comment]
I don't have that much knowledge of Ci, so you or Someone else have to do.

**Comment by Trancever** [comment]
@erandagan @Suraj-Tiwari I published 6.0.0-rc.3 with latest fixes.
